libs/hmpb/src/ballot_styles.ts

```diff
@@ -68,28 +82,48 @@ export function generateBallotStyles(params: {
     // Remove districtIds after grouping, we don't need them anymore
     group.map(({ precinctId, splitId }) => ({ precinctId, splitId })),
   ]);
+  let ballotStyleIndex = 0;
 
   switch (electionType) {
     case 'general':
       return precinctsOrSplitsByDistricts.flatMap(
-        ([districtIds, precinctsOrSplits], index) =>
-          ballotLanguageConfigs.map(({ languages }) => ({
-            id: generateBallotStyleId({
-              ballotStyleIndex: index + 1,
-              languages,
-            }),
-            groupId: generateBallotStyleGroupId({
-              ballotStyleIndex: index + 1,
-            }),
-            precincts: precinctsOrSplits.map(({ precinctId }) => precinctId),
-            districts: districtIds,
-            languages,
-          }))
+        ([districtIds, precinctsOrSplitIds]) => {
+          // Get all ordered contest sets for this district group
+          const orderedContestSets = getAllPossibleCandidateOrderings(
+            ballotTemplateId,
+            {
+              contests,
+              precincts,
+              districtIds,
+              precinctsOrSplitIds,
+              electionId,
+            }
+          );
+          // Create ballot styles for each ordered contest set and language config
+          return orderedContestSets.flatMap(
+            ({ orderedContests, precinctsOrSplits: precinctsForSet }) => {
+              ballotStyleIndex += 1;
```

Comment:
I wonder if we should generate ballot style IDs that help us see which ballot styles have the same contests with a different rotation? E.g. if `1_en`, `2_en` have the same contests with different rotation, maybe we name them `1.1_en` and `1.2_en` or something